src/bots/unslashed-kleros-connector/withdraw-fees-and-rewards.js

```diff
@@ -0,0 +1,91 @@
+const delay = require('delay')
+
+const Ruling = {
+  None: 0,
+  Claimant: 1,
+  Challenger: 2
+}
+const klerosConnectorData = JSON.parse(process.env.UNSLASHED_KLEROS_CONNECTOR)
+const deploymentBlock = klerosConnectorData.blockNumber
+
+module.exports = async (web3, batchedSend, klerosConnector) => {
+  // Keep track of the disputes so that we don't waste unnecessary resources on them.
+  const withdrawnDisputeIDs = {}
+  const queryFrequency = 60 // minutes
+
+  while (true) {
+    const transactionList = []
+
+    // Loop over all disputes.
+    try {
+      let localDisputeID = 0
+      while (true) {
+        if (withdrawnDisputeIDs[localDisputeID]) {
+          // Fees already withdrawn.
+          localDisputeID++
+          continue
+        }
+
+        const disputeData = await klerosConnector.methods.disputes(localDisputeID).call()
+        if (!disputeData.resolved) {
+          // Dispute is not resolved yet.
+          localDisputeID++
+          continue
+        }
+
+        // Get contribution events.
+        const contributionEvents = await klerosConnector.getPastEvents(
+          'Contribution',
+          {
+            fromBlock: deploymentBlock,
+            filter: {
+              localDisputeID: web3.utils.toBN(localDisputeID)
+            }
+          }
+        )
+
+        const disputeWithdrawals = []
+        for (let i = 0; i < contributionEvents.length; i++) {
+          const contribution = contributionEvents[i].returnValues
+          if (contribution.round == 0) continue // Round zero is automatically withdrawn by the contract logic.
+
+          if (disputeData.ruling == Ruling.None || contribution.ruling == disputeData.ruling) {
```

Comment:
It doesn't seem to take into account the unsuccessful fundings. If you were funding the losing side but didn't fund it fully you're still eligible to reimbursement.
https://gitlab.com/UnslashedFinance/claims/-/blob/dev/contracts/KlerosConnector.sol#L592

Reply:
Good catch! I'll fix that.

Reply:
Fixed.